src/lightning_app/storage/mount.py

```diff
@@ -42,8 +43,14 @@ def __post_init__(self) -> None:
                 f"Received: '{self.source}'. Mounting a single file is not currently supported."
             )
 
-        if self.root_dir == "":
-            self.root_dir = f"/data/{Path(self.source).stem}"
+        if self.mount_path == "":
+            self.mount_path = f"/data/{Path(self.source).stem}"
+
+        if not os.path.isabs(self.mount_path):
```

Comment:
good call, maybe add a test for this condition?

Reply:
I already did! 

https://github.com/Lightning-AI/lightning/pull/15228/files#diff-6a0b6976b768aa8a7585702b656db550b2b1d997b8c4f765707d1dd2589cf986R40-R42